### ISSUE TITLE
[Bug?]: Static assets with `vercel_static` preset are incorrect

### ISSUE THREAD

**Issue by oscartbeaumont** [solidjs/solid-start#1309]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

The static files in the `.vercel/output/static` directory are lacking a proper HTML file extension.

This causes them to be treated as a download instead of serving them as HTML.

### Expected behavior 🤔

The HTML page should be shown instead of presenting a browser download prompt.

### Steps to reproduce 🕹

Steps:

1. `pnpm create solid` (SSR & TS enabled, using base template)
2. `NITRO_PRESET=vercel_static pnpm build`
3. Observe the `.vercel/output/static` directory. See `index` not `index.html`

### Workaround

I found this hack that works around the issue for now:

```js
start: {
  server: {
    prerender: {
      // We don't do this to force output from `index` to `index.html`
      routes: ["/index.html"],
    },
  },
},
```

This also require renaming my file from `index.*` to `index.html.*`.

**Comment by birkskyum** [comment]
I tried this today, and I believe I got a 'correct' output, as shown below. Just to clarify, the solid template you used, is it `bare` or `basic`? You wrote `base`, which is ambiguous.

<img width="458" alt="Screenshot 2024-02-12 at 18 24 53" src="https://github.com/solidjs/solid-start/assets/74932975/00532f04-ce9d-4ff9-86c7-8d0fdf898b96">

**Comment by ryansolid** [comment]
Sorry this was a bug I fixed in v0.5.4. I missed setting the right headers.
